### ISSUE TITLE
Detecting clicks outside of a tooltip – `handleOutsideClick` fires after unmounting (IE only)

### ISSUE THREAD

**Issue by olekbruks** [mohsinulhaq/react-popper-tooltip#40]
I’ve found a very specific scenario in which `handleOutsideClick` function is fired when the `Tooltip` component is unmounted. 

In my web app (using `react`, `redux` and `react-redux-router`) I have a button which acts as a tooltip trigger (trigger mode is set to “click”) and a button which executes `history.push`. Due to the routing, navigating through the app using “push” won’t reenter the app but only unmounts some components and mounts another. In IE 11 when the user clicks on the second button when the tooltip is shown react unmounts parent component, buttons and hence the `TooltipTrigger` and the `Tooltip` components. The `handleOutsideClick` event handler is removed, but (only in IE) the code in the handler is executed which leads to exception because `findDOMNode` cannot be used on unmounted components.

In my opinion, the simplest solution is to check whether the component is mounted before calling `findDOMNode`. I don't have any idea why the handler is called after removing the event listener.

**Comment by mohsinulhaq** [comment]
I am not sure about what exactly is happening here. Can you attach a gif of the actions you take?
If you are unable to do that, I want to ask if the button that does the routing is inside a tooltip. And does the whole page unmount with route change or is it some fragment routing setup.

**Comment by olekbruks** [comment]
I will try to replicate this behavior in an online editor/sandbox. 

In meantime: the button which does the routing is outside a tooltip, in a component that will be unmounting by a route change. The route dependent component is just a "body" of web application surrounded by a top bar and a navigation box.

- App
   - Page component 1
      - Button with tooltip
      - Button which changes routing to "Page component 2"
   - Page component 2

**Comment by olekbruks** [comment]
@mohsinulhaq Any updates on this? I think that the solution (to check whether the component is mounted before calling `findDOMNode`) is good enough. 

BTW I'm unable to build the project on Windows - after build the file `dist\cjs\react-popper-tooltip.js` contains only a few lines.

**Comment by mohsinulhaq** [comment]
I see that the issue can also be reproduced in Edge. Will check that this weekend.
And about the build. Can you paste the build log here?

**Comment by olekbruks** [comment]
Indeed, it's broken in the Edge too.

After some changes in `package.json` and `rollup.config.js` I managed to fix some issues and now rollup doesn't signal that something went wrong. However the output file - `dist/cjs/react-popper-tooltip.js` is almost empty:

```js
'use strict';

var TooltipTrigger = require('./TooltipTrigger.tsx');

module.exports = TooltipTrigger;
//# sourceMappingURL=react-popper-tooltip.js.map
```

Terminal output:

```bash
> rimraf dist && rollup -c && xcopy src\styles.css dist\* /y && yarn tsc && rimraf compiled

src/index.ts → dist/cjs/react-popper-tooltip.js...
created dist/cjs/react-popper-tooltip.js in 1.1s

src/index.ts → dist/esm/react-popper-tooltip.js...
created dist/esm/react-popper-tooltip.js in 44ms
src\styles.css
1 File(s) copied
yarn run v1.13.0
$ C:\Users\\source\repos\react-popper-tooltip\node_modules\.bin\tsc
Done in 10.39s.
```

**Comment by mohsinulhaq** [comment]
Hi, I have released v2.8.1-alpha.1 with a check for mount state before findDOMNode on outside click. Please give it a spin and let me know if anything is missing.

**Comment by olekbruks** [comment]
The issue is resolved in the alpha version.

**Comment by mohsinulhaq** [comment]
Thanks!

**Comment by mohsinulhaq** [comment]
Fix now available in [v2.8.1](https://github.com/mohsinulhaq/react-popper-tooltip/releases/tag/v2.8.1)
